src/Arcus.Security.Core/SecretStoreBuilder.cs

```diff
@@ -40,19 +42,35 @@ public SecretStoreBuilder(IServiceCollection services)
         /// <remarks>
         ///     The series of secret stores is directly publicly available including the operations so future (consumer) extensions can easily manipulate this series during build-up.
         /// </remarks>
-        public IList<SecretStoreSource> SecretStoreSources { get; } = new List<SecretStoreSource>();
+        public IList<Func<IServiceProvider, SecretStoreSource>> CreateSecretStoreSources { get; } = new List<Func<IServiceProvider, SecretStoreSource>>();
```

Comment:
Had to change this to have access to the `IServiceProvider` when initiating the `MutatedSecretNameSecretProvider` instance. Technically, this is a breaking change bc it's public, but not sure if it was needed to be public in the first place plus it was/will only be used when you want to manipulated already registered secret sources (so we may not allow that anyway).

Reply:
So, I could make this backwards compatible with obsolete properties or I could leave it as it is.

Reply:
Ah, I could just extend the `SecretSource` itself. hang on...

Reply:
Ok, nevermind this. Changed it.